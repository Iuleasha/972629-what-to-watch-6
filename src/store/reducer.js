import {AuthorizationStatus, DEFAULT_GENRE} from '../constant';
import {createGenreArray} from '../filter/filter';
import {ActionType} from './action';

const initialState = {
  genre: DEFAULT_GENRE,
  genres: [],
  films: [],
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  isDataLoaded: false,
  comments: {},
  preview: null,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_GENRE:
      return {
        ...state,
        genre: action.payload,
      };

    case ActionType.LOAD_SUCCESS:
      return {
        ...state,
        genres: createGenreArray(action.payload),
        films: action.payload,
        isDataLoaded: true,
      };

    case ActionType.LOAD_PROMO_FILM:
      return {
        ...state,
        preview: action.payload,
      };

    case ActionType.REQUIRED_AUTHORIZATION:
      return {
        ...state,
        authorizationStatus: action.payload,
      };

    case ActionType.SET_USER:
      return {
        ...state,
        user: action.payload,
      };

    case ActionType.ADD_REVIEW:
      const {comments} = state;

      return {
        ...state,
        comments: {...comments, [action.payload.id]: action.payload.comments},
      };

    case ActionType.LOAD_FAVORITE:
      return {
        ...state,
        favorite: action.payload,
      };

    case ActionType.SWITCH_FAVORITE_STATUS:
      if (state.preview && state.preview.id === action.payload.id) {
        state.preview = action.payload;
      }

      state.films = state.films.map((item) => item.id === action.payload.id ?
        action.payload : item
      );

      return {
        ...state,
        favorite: state.films.filter(({isFavorite}) => isFavorite),
      };
  }

  return state;
};


export {reducer};
