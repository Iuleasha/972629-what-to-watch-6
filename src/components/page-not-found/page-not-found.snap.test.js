import {render} from '@testing-library/react';
import {createMemoryHistory} from 'history';
import React from 'react';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import {createStore} from 'redux';
import reducer from '../../store/root-reducer';
import PageNotFound from './page-not-found';

const store = createStore(
    reducer,
);

test(`Should PageNotFoundScreen render correctly`, () => {
  const history = createMemoryHistory();
  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <PageNotFound/>
        </Router>
      </Provider>,
  );
  expect(container).toMatchSnapshot();
});
