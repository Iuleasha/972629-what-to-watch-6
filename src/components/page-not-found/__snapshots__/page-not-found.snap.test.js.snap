// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PageNotFoundScreen render correctly 1`] = `
<div>
  <div
    class="user-page"
  >
    <header
      class="page-header user-page__head"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="user-block"
      >
        <a
          class="user-block__link"
          href="/login"
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      class="sign-in user-page__content"
    >
      <h1>
        404 Not Found
      </h1>
      <a
        href="/"
      >
        Вернуться на главную страницу
      </a>
    </div>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link logo__link--light"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          © 
          2021
           What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
